### PR TITLE
Mepts 1069

### PR DESCRIPTION
**Pull request by ningosi**
https://jembiprojects.jira.com/browse/MEPTS-1069

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListChildrenAdolescentARTWithoutFullDisclosureCohortQueries.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    cd.setQuery(
+        "SELECT p.patient_id FROM patient p "
+            + " INNER JOIN( "
+            + query
+            + ") art ON p.patient_id=art.patient_id"
+            + " WHERE p.voided=0 AND art.first_pickup IS NOT NULL ");
+    return cd;
+  }
+
+  public CohortDefinition getAdolescentsCurrentlyOnArtWithDisclosures(int valueCoded) {
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListChildrenAdolescentARTWithoutFullDisclosureCohortQueries.java:101]: @ningosi can you please put the spec here

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListChildrenAdolescentARTWithoutFullDisclosureCohortQueries.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            + " SELECT p.patient_id,MAX(e.encounter_datetime) AS encounter_datetime FROM patient p "
+            + " INNER JOIN encounter e ON p.patient_id=e.patient_id "
+            + " INNER JOIN obs o ON e.encounter_id=o.encounter_id "
+            + " WHERE p.voided=0 AND e.voided=0 AND o.voided=0 AND e.encounter_type = ${35} "
+            + " AND o.concept_id=${6340} AND e.encounter_datetime <= :endDate "
+            + " AND e.location_id=:location GROUP BY p.patient_id) tt ON p.patient_id=tt.patient_id "
+            + " INNER JOIN encounter e1 ON p.patient_id=e1.patient_id "
+            + " INNER JOIN obs ob ON e1.encounter_id=ob.encounter_id "
+            + " WHERE tt.encounter_datetime=e1.encounter_datetime AND p.voided=0 "
+            + " AND e1.voided=0 AND ob.voided=0 AND ob.value_coded= ${answer} ";
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListChildrenAdolescentARTWithoutFullDisclosureCohortQueries.java:125]: I think here is missing the encounter_type and location....

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListChildrenAdolescentARTWithoutFullDisclosureCohortQueries.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+    StringSubstitutor sb = new StringSubstitutor(map);
+    String replacedQuery = sb.replace(query);
+    cd.setQuery(replacedQuery);
+    return cd;
+  }
+
+  public CohortDefinition getTotalAdolescentsCurrentlyOnArtWithBlankDisclosures() {
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListChildrenAdolescentARTWithoutFullDisclosureCohortQueries.java:138]: Spec

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListChildrenAdolescentARTWithoutFullDisclosureCohortQueries.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            + " AND o.concept_id=${6340} AND e.encounter_datetime <= :endDate "
+            + " AND e.location_id=:location)";
+
+    StringSubstitutor sb = new StringSubstitutor(map);
+    String replacedQuery = sb.replace(query);
+    cd.setQuery(replacedQuery);
+    return cd;
+  }
+
+  public CohortDefinition getAdolescentsCurrentlyOnArtWithoutDisclosures(int valueCoded) {
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListChildrenAdolescentARTWithoutFullDisclosureCohortQueries.java:170]: Spec

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenAdolescentARTWithoutFullDisclosureDataset.java [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+        hivMetadata.getDisclosureOfHIVDiagnosisToChildrenAdolescentsConcept());
+    obsForPersonDataDefinition.setWhich(TimeQualifier.LAST);
+    obsForPersonDataDefinition.setValueCodedList(
+        Arrays.asList(
+            hivMetadata.getPartiallyRevealedConcept(), hivMetadata.getNotRevealedConcept()));
+
+    return obsForPersonDataDefinition;
+  }
+
+  private DataDefinition getAdolescentsCurrentlyOnArtWithPartialDisclosures(Concept valueCoded) {
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenAdolescentARTWithoutFullDisclosureDataset.java:163]: Spec

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenAdolescentARTWithoutFullDisclosureDataset.java [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+            + " GROUP BY p.patient_id";
+
+    StringSubstitutor sb = new StringSubstitutor(map);
+    String replacedQuery = sb.replace(query);
+    cd.setQuery(replacedQuery);
+    return cd;
+  }
+
+  private DataDefinition
+      getAdolescentsCurrentlyOnArtWithPartialDisclosuresDifferenceFromReportingDate(
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenAdolescentARTWithoutFullDisclosureDataset.java:190]: Spec

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
@ningosi please have a look

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListChildrenAdolescentARTWithoutFullDisclosureCohortQueries.java (hunk 2 above)
Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListChildrenAdolescentARTWithoutFullDisclosureCohortQueries.java:125]: @ningosi here it's missing the encounter type

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenAdolescentARTWithoutFullDisclosureDataset.java (hunk 6 above)
Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenAdolescentARTWithoutFullDisclosureDataset.java:190]: @ningosi how do you know they Adolescents, is it checked somewhere else?

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
@ningosi missing one comment to be addressed

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenAdolescentARTWithoutFullDisclosureDataset.java (hunk 6 above)
Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenAdolescentARTWithoutFullDisclosureDataset.java:190]: Those are filtered in the base cohort passed. Patients who are between 8 to 14 years see this getBaseCohortForAdolescent() method that combines all

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**